nav2_collision_monitor/src/velocity_polygon.cpp

```diff
@@ -0,0 +1,195 @@
+      // polygon points parameter
+      std::vector<Point> poly;
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name_ + "." + velocity_polygon_name + ".points", rclcpp::PARAMETER_STRING);
+      std::string poly_string =
+        node->get_parameter(polygon_name_ + "." + velocity_polygon_name + ".points").as_string();
+
+      if (!getPolygonFromString(poly_string, poly)) {
+        return false;
+      }
+
+      // linear_min param
+      double linear_min;
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name_ + "." + velocity_polygon_name + ".linear_min",
+        rclcpp::ParameterValue(0.0));
+      linear_min = node->get_parameter(polygon_name_ + "." + velocity_polygon_name + ".linear_min")
+        .as_double();
+
+      // linear_max param
+      double linear_max;
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name_ + "." + velocity_polygon_name + ".linear_max",
+        rclcpp::ParameterValue(0.0));
+      linear_max = node->get_parameter(polygon_name_ + "." + velocity_polygon_name + ".linear_max")
+        .as_double();
+
+      // theta_min param
+      double theta_min;
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name_ + "." + velocity_polygon_name + ".theta_min",
+        rclcpp::ParameterValue(0.0));
+      theta_min =
+        node->get_parameter(polygon_name_ + "." + velocity_polygon_name + ".theta_min").as_double();
+
+      // theta_max param
+      double theta_max;
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name_ + "." + velocity_polygon_name + ".theta_max",
+        rclcpp::ParameterValue(0.0));
+      theta_max =
+        node->get_parameter(polygon_name_ + "." + velocity_polygon_name + ".theta_max").as_double();
+
+      // direction_end_angle param and direction_start_angle param
+      double direction_end_angle = 0.0;
+      double direction_start_angle = 0.0;
+      if (holonomic_) {
+        nav2_util::declare_parameter_if_not_declared(
+          node, polygon_name_ + "." + velocity_polygon_name + ".direction_end_angle",
+          rclcpp::ParameterValue(M_PI));
+        direction_end_angle =
+          node->get_parameter(polygon_name_ + "." + velocity_polygon_name + ".direction_end_angle")
+          .as_double();
+
+        nav2_util::declare_parameter_if_not_declared(
+          node, polygon_name_ + "." + velocity_polygon_name + ".direction_start_angle",
+          rclcpp::ParameterValue(-M_PI));
+        direction_start_angle =
+          node
+          ->get_parameter(polygon_name_ + "." + velocity_polygon_name + ".direction_start_angle")
+          .as_double();
+      }
+
+      SubPolygonParameter sub_polygon = {
+        poly, velocity_polygon_name, linear_min, linear_max, theta_min,
+        theta_max, direction_end_angle, direction_start_angle};
+      sub_polygons_.emplace_back(sub_polygon);
```

Comment:
You can emplace back with ` poly, velocity_polygon_name, linear_min, linear_max, theta_min,
        theta_max, direction_end_angle, direction_start_angle`. There's no need for the object instantiation